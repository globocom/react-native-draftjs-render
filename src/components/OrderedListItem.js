/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

// @flow

import React from 'react';
import {
  Text,
  View,
  StyleSheet,
} from 'react-native';

import DraftJsText from '../components/DraftJsText';

import type { OrderedListItemPropsType } from './types';

const styles = StyleSheet.create({
  orderedListItemContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  orderedListItemNumber: {
    fontSize: 12,
    fontWeight: 'bold',
    marginRight: 8,
    alignSelf: 'center',
  },
});

const OrderedListItem = (props: OrderedListItemPropsType): any => {
  const {
    counter,
    separator,
  } = props;


  const orderedListItemCustomStyleContainer = props.customStyles ?
    props.customStyles.orderedListItemContainer :
    undefined;

  const orderedListItemCustomStyleNumber = props.customStyles ?
    props.customStyles.orderedListItemNumber :
    undefined;

  let marginLeft = 0;
  marginLeft =
      orderedListItemCustomStyleNumber && orderedListItemCustomStyleNumber.marginLeft ?
        props.depth * orderedListItemCustomStyleNumber.marginLeft :
        props.depth * props.defaultMarginLeft;

  return (
    <View style={[styles.orderedListItemContainer, orderedListItemCustomStyleContainer]}>
      <Text
        style={[styles.orderedListItemNumber, orderedListItemCustomStyleNumber,
          { marginLeft }]}
      >
        {counter}{separator}
      </Text>
      <DraftJsText
        {...props}
      />
    </View>);
};

OrderedListItem.defaultProps = {
  counter: 1,
  depth: 0,
  separator: '.',
  defaultMarginLeft: 8,
};

export default OrderedListItem;
