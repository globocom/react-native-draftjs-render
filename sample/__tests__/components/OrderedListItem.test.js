/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

/* eslint-env jest */

import 'react-native';

import React from 'react';
import renderer from 'react-test-renderer';

import OrderedListItem from '../../../src/components/OrderedListItem';

it('renders correctly with a ordered-list-item', () => {
  const text = 'Hello World';
  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with a ordered-list-item when depth is undefined', () => {
  const text = 'Hello World';
  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    depth={undefined}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with a ordered-list-item when depth 1', () => {
  const text = 'Hello World';
  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    depth={1}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when customStyles is undefined', () => {
  const text = 'Hello World';
  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    customStyles={undefined}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders null without a ordered-list-item', () => {
  const tree = renderer.create(<OrderedListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('extends a style with customStyles', () => {
  const text = 'Hello World';
  const customStyles = {
    'ordered-list-item': {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
      marginLeft: 10,
    },
  };

  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('extends a style with a customStyle.orderedListItemContainer', () => {
  const text = 'Hello World';
  const customStyles = {
    'ordered-list-item': {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
      marginLeft: 10,
    },
    orderedListItemContainer: {
      flex: 2,
    },
  };
  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('extends a style with a customStyle.orderedListItemNumber', () => {
  const text = 'Hello World';
  const customStyles = {
    'ordered-list-item': {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
      marginLeft: 10,
    },
    orderedListItemNumber: {
      fontSize: 14,
    },
  };
  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when orderedListItemNumber.marginLeft is set', () => {
  const text = 'Hello World';
  const customStyles = {
    'ordered-list-item': {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
      marginLeft: 10,
    },
    orderedListItemNumber: {
      marginLeft: 14,
    },
  };
  const tree = renderer.create(<OrderedListItem
    type="ordered-list-item"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
