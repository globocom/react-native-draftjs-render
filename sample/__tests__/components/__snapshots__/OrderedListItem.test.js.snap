// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extends a style with a customStyle 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "flex": 2,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 12,
          "fontWeight": "bold",
          "marginRight": 8,
        },
        Object {
          "fontSize": 14,
        },
        Object {
          "marginLeft": 0,
        },
      ]
    }
  >
    1
    .
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        Object {
          "fontSize": 18,
          "fontWeight": "normal",
          "letterSpacing": -0.75,
          "lineHeight": 32,
          "marginLeft": 10,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders correctly with a ordered-list-item 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 12,
          "fontWeight": "bold",
          "marginRight": 8,
        },
        undefined,
        Object {
          "marginLeft": 0,
        },
      ]
    }
  >
    1
    .
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders correctly with a ordered-list-item when depth 1 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 12,
          "fontWeight": "bold",
          "marginRight": 8,
        },
        undefined,
        Object {
          "marginLeft": 7,
        },
      ]
    }
  >
    1
    .
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders correctly with a ordered-list-item when depth is undefined 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 12,
          "fontWeight": "bold",
          "marginRight": 8,
        },
        undefined,
        Object {
          "marginLeft": 0,
        },
      ]
    }
  >
    1
    .
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders null without a ordered-list-item 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 12,
          "fontWeight": "bold",
          "marginRight": 8,
        },
        undefined,
        Object {
          "marginLeft": 0,
        },
      ]
    }
  >
    1
    .
  </Text>
</View>
`;
